client/coral-embed-stream/src/CommentStream.js

```diff
@@ -284,20 +279,14 @@ const mapStateToProps = state => ({
   userData: state.user.toJS()
 });
 
-const mapDispatchToProps = (dispatch) => ({
-  addItem: (item, itemType) => dispatch(addItem(item, itemType)),
-  updateItem: (id, property, value, itemType) => dispatch(updateItem(id, property, value, itemType)),
-  postItem: (data, type, id) => dispatch(postItem(data, type, id)),
-  getStream: (rootId) => dispatch(getStream(rootId)),
-  addNotification: (type, text) => dispatch(addNotification(type, text)),
-  clearNotification: () => dispatch(clearNotification()),
-  showSignInDialog: (offset) => dispatch(showSignInDialog(offset)),
-  postAction: (item, action, user, itemType) => dispatch(postAction(item, action, user, itemType)),
-  deleteAction: (item, action, user, itemType) => dispatch(deleteAction(item, action, user, itemType)),
-  appendItemArray: (item, property, value, addToFront, itemType) => dispatch(appendItemArray(item, property, value, addToFront, itemType)),
-  handleSignInDialog: () => dispatch(authActions.showSignInDialog()),
-  logout: () => dispatch(logout()),
-  updateStatus: status => dispatch(updateOpenStatus(status))
-});
+// Ties all imported actions to dispatch and makes them available in props.
+const mapDispatchToProps = (dispatch) => {
+  const actions = Object.assign({}, itemActions, notificationActions, authActions, configActions);
+
+  return Object.keys(actions).reduce((dispatchActions, key) => {
```

Comment:
I don't think we should do this. Adding another iteration attempts to the performance. And the reason behind `mapDispatchToProps` is that we are 100% aware of what our component can do. 
Something that we can do to make this file smaller is abstracting the functionality into smaller components. This Component is making a lot of things and it shouldn't. 

 Breaking this into smaller components like we have with the `coral-sign-in` `coral-settings` or `coral-configure`, in my opinion, is the way to go.